day1.scala

```diff
@@ -0,0 +1,26 @@
+// using scala 3.0.2
+
+package day1
+
+import scala.io.Source
+
+@main def part1(): Unit =
+  val input = Source.fromFile("input/day1").mkString
+  val answer = part1(input)
+  println(s"The solution is $answer")
+
+@main def part2(): Unit =
+  val input = Source.fromFile("input/day1").mkString
+  val answer = part2(input)
+  println(s"The solution is $answer")
+
+def part1(input: String): Int = 
+  val depths = input.linesIterator.map(_.toInt)
+  val pairs = depths.sliding(2).map(arr => (arr(0), arr(1)))
+  pairs.count((prev, next) => prev < next)
+
+def part2(input: String): Int =
+  val depths = input.linesIterator.map(_.toInt)
```

Comment:
so don't you prefer to read and parse the file on the previous level, so this method cannot file (I mean function part1 and part2, would take seq[INT] and return an Int..

Reply:
it's just a suggestion .. it wont change the code

Reply:
yes it would be better

the reason I did otherwise is that the web solver (in the website) takes the raw input as parameter

Reply:
forgot about it!